src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs

```diff
@@ -3,15 +3,18 @@
 
 using System;
 using Microsoft.AspNet.Mvc.ViewFeatures;
+using System.Collections.Generic;
+using Microsoft.Extensions.WebEncoders;
 
 namespace Microsoft.AspNet.Mvc.Rendering
 {
     public class MvcForm : IDisposable
     {
         private readonly ViewContext _viewContext;
+        private readonly IHtmlEncoder _htmlEncoder;
         private bool _disposed;
 
-        public MvcForm(ViewContext viewContext)
+        public MvcForm(ViewContext viewContext, IHtmlEncoder htmlEncoder)
```

Comment:
Ick. For the rare case when the `ViewContext.TextWriter` isn't an `HtmlTextWriter`, use `ViewContext.HttpContext.RequestServices.GetRequiredService<IHtmlEncoder>()`.